src/utils/components/SSHSecretSection/utils/utils.ts

```diff
@@ -169,3 +193,49 @@ export const getPropagationMethod = (
   vm: V1VirtualMachine,
 ): V1SSHPublicKeyAccessCredentialPropagationMethod =>
   vm?.spec?.template?.spec?.accessCredentials?.[0].sshPublicKey.propagationMethod;
+
+const generateShortPrettyName = (prefix?: string): string => {
+  const numberDictionary = NumberDictionary.generate({ length: 2 });
+  const prefixValue = prefix ? `${prefix}-` : '';
+
+  return `${prefixValue}${uniqueNamesGenerator({
+    dictionaries: [animals, numberDictionary],
+    separator: '-',
+  })}`;
+};
+
+const generateMinimalPrettyName = (prefix?: string): string => {
+  const numberDictionary = NumberDictionary.generate({
+    length: MAX_NAME_LENGTH - HYPHEN_LENGTH - prefix.length,
+  });
+
+  return `${prefix}-${numberDictionary}`;
+};
+
+export const generateValidSecretName = (
+  secretName: string,
+  vmNamespaceTarget: string,
+  secrets: IoK8sApiCoreV1Secret[],
+) => {
+  const name =
+    secretName.length > MIN_NAME_LENGTH_FOR_GENERATED_SUFFIX
+      ? secretName.substring(0, MIN_NAME_LENGTH_INDEX)
+      : secretName;
+
+  let isValid = false;
+  let generatedName = '';
+
+  do {
+    const nameLength = name.length;
+
+    if (nameLength <= SAFE_NAME_LENGTH_FOR_NORMAL_GENERATED_SUFFIX)
+      generatedName = generatePrettyName(name);
+    else if (nameLength <= SAFE_NAME_LENGTH_FOR_SHORT_GENERATED_SUFFIX)
+      generatedName = generateShortPrettyName(name);
+    else generatedName = generateMinimalPrettyName(name);
+
+    isValid = validateSecretName(generatedName, vmNamespaceTarget, secrets);
+  } while (!isValid);
+
+  return generatedName;
+};
```

Comment:
Why do all of this, instead of checking if it's above 70 chars, if yes, generate a new name?

Reply:
I think the max is 51 chars. Isn't it?

Reply:
51 or 70 or 100 isnt really matter, should check the limit :)

Reply:
@metalice @hstastna I was trying to keep as much of the original secret name as possible while still adding a suffix to making it unique, but I'm good with simplifying it. 

The name limit is 51 characters, which includes any suffix we add. I generated 5,000 suffixes multiple times. A few times I got a single 9 character suffix, but I always got a few 10 character suffixes, so I think we should use 10 characters as the minimum suffix length.

With that we know that every existing secret name greater than 41 characters will be too long to get a suffix and remain under the character limit. So, are both of you good with the follow algorithm?

if (secretName.length > 41) generate a totally random name
else generate and attach a suffix

Reply:
yes i think its much better to generate a new name if > 41

Reply:
I am not sure, for me this all seems to be too complicated unnecessary. I'd just generate a suffix as originally and if the final name longer than expected, I'd just cut from the suffix what is extra to achieve being the final name max 51 chars. Or.. or I'd just simply generate a totally random name always, no matter the original secret name. WDYT @pcbailey @metalice ?

Reply:
I just pushed the updated logic, but I'm fine with either of those if that's what you guys want. 

FYI, the logic I just pushed will ensure that names that can get a suffix will get one that fits and creates a valid name while those too long for a suffix get a newly generated name.